src/components/AddressSummaryCard.tsx

```diff
@@ -75,52 +73,76 @@ export default styled(AddressSummaryCard)`
   flex-direction: column;
   width: ${addressSummaryCardWidthPx}px;
   height: 95%;
-  background-color: ${({ theme }) => theme.bg.secondary};
-  border: 1px solid ${({ theme }) => theme.border.secondary};
-  border-radius: var(--radius-medium);
-`
 
-const PaddedSection = styled.div`
-  padding: 11px 17px;
+  ${({ clickable }) =>
+    !clickable &&
+    css`
+      background-color: ${({ theme }) => theme.bg.secondary};
+      border-radius: var(--radius-medium);
+      border: 1px solid ${({ theme }) => theme.border.primary};
+    `}
 `
 
-const AddressNameSection = styled(PaddedSection)`
+const AddressNameSection = styled.div<{ collapsed: boolean }>`
   display: flex;
   flex-direction: column;
   align-items: center;
-
-  > * {
-    max-width: 100%;
-  }
+  transition: padding 0.2s ease-out;
+  padding: 0 17px;
+
+  ${({ collapsed }) =>
+    !collapsed &&
+    css`
+      background-color: ${({ theme }) => theme.bg.secondary};
+      border-top-left-radius: var(--radius-medium);
+      border-top-right-radius: var(--radius-medium);
+      padding: 0;
+    `}
 `
 
-const Hash = styled(Truncate)`
-  color: ${({ theme }) => theme.font.secondary};
-  font-size: 10px;
-  font-weight: var(--fontWeight-medium);
-  margin-top: var(--spacing-1);
+const AddressBadgeStyled = styled(AddressBadge)`
+  width: 100%;
+  padding: 11px 17px;
+  border-radius: var(--radius-medium);
+  border-bottom-left-radius: 0;
+  border-bottom-right-radius: 0;
```

Comment:
You could use :
```css
border-radius: var(--radius-medium) var(--radius-medium) 0 0;
```

or better yet, `border-radius: 0` with `ovrflow: hidden` on the parent `AddressNameSection`.

WDYT?

Reply:
That totally works! And it feels much cleaner, thanks! Pushing...